SwiftGit2/Repository.swift

```diff
@@ -537,4 +537,149 @@ final public class Repository {
 		let iterator = CommitIterator(repo: self, root: branch.oid.oid)
 		return iterator
 	}
+
+	// MARK: - Diffs
+
+	public func diff(for commit: Commit) -> Result<[Diff.Delta], NSError> {
+		return self
+			.commit(with: commit.oid)
+			.flatMap { baseCommit in
+				return self.tree(from: baseCommit)
+			}
+			.flatMap { baseTree in
+				guard !commit.parents.isEmpty else {
+					// Initial commit in a repository
+					return self.diff(withOldTree: nil, andNewTree: baseTree)
+				}
+
+				var mergeDiff: Result<OpaquePointer?, NSError> = .success(nil)
+				for parent in commit.parents {
+					mergeDiff = mergeDiff
+						.flatMap { mergeDiff in
+							return self
+								.parentCommit(from: parent)
+								.flatMap { commit in
+									return self.tree(from: commit)
+								}
+								.flatMap { tree in
+									return self.diff(withOldTree: tree, andNewTree: baseTree)
+								}
+								.flatMap { diff in
+									guard let mergeDiff = mergeDiff else { return .success(diff) }
+									let mergeResult = git_diff_merge(mergeDiff, diff)
+									guard mergeResult == GIT_OK.rawValue else {
+										return .failure(NSError(gitError: mergeResult, pointOfFailure: "git_diff_merge"))
+									}
+									return .success(mergeDiff)
+							}
+					}
+				}
+				return .success(mergeDiff.value!!)
+			}
+			.flatMap { diffResult in
+				return self.processDiffDeltas(diffResult)
+		}
+	}
+
+	private func commit(with oid: OID) -> Result<OpaquePointer, NSError> {
+		var unsafeBaseCommit: OpaquePointer? = nil
+		let unsafeBaseOid = UnsafeMutablePointer<git_oid>.allocate(capacity: 1)
+		git_oid_fromstr(unsafeBaseOid, oid.description)
+		let lookupBaseGitResult = git_commit_lookup(&unsafeBaseCommit, self.pointer, unsafeBaseOid)
+		guard lookupBaseGitResult == GIT_OK.rawValue, let unwrapBaseCommit = unsafeBaseCommit else {
+			return Result.failure(NSError(gitError: lookupBaseGitResult, pointOfFailure: "git_commit_lookup"))
+		}
+		git_commit_free(unsafeBaseCommit)
+
+		return Result.success(unwrapBaseCommit)
+	}
+
+	private func diff(withOldTree oldTree: OpaquePointer?,
+	                  andNewTree newTree: OpaquePointer?) -> Result<OpaquePointer, NSError> {
```

Comment:
It doesn't matter that much since this is private, but I think `diff(fromTree:toTree:)` would be more idiomatic labels.